app/app/explore/components/AddLiquidityModal.tsx

```diff
@@ -0,0 +1,274 @@
+                {getAssetShortSymbol(pool.asset)} Balance:{" "}
+                {isERC20(pool.asset) ? assetBalance.toFixed(6) : "N/A"}
+              </span>
+            </div>
+            <div>
+              {assetAmount.toPrecision(6)} ($
+              {(parseFloat(pool.assetPriceUSD) * assetAmount).toFixed(2)})
+            </div>
+          </div>
+
+          <div className="relative mb-6">
+            <Slider
+              value={assetAmount}
+              max={assetBalance}
+              onChange={setAssetAmount}
+            />
+          </div>
+
+          <div className="flex justify-end gap-2">
+            {[25, 50, 100].map((percent) => (
+              <button
+                key={percent}
+                className={percentageButtonClasses(
+                  isCloseToPercentage(currentAssetPercentage, percent),
+                )}
+                onClick={() => handlePercentageClick(percent)}
+              >
+                {percent === 100 ? "MAX" : `${percent}%`}
+              </button>
+            ))}
+          </div>
+        </div>
+
+        {/* RUNE Input Section */}
+        {selectedTab === "double" && (
+          <div className="mt-6">
+            <div className="flex items-center justify-between mb-4 text-neutral text-2xl font-medium">
+              <div className="flex items-center">
+                <Image
+                  src={getLogoPath("thor.rune")}
+                  alt="RUNE"
+                  width={42}
+                  height={42}
+                  className="mr-3"
+                />
+                <span className="font-gt-america-ext">
+                  RUNE Balance: {runeBalance.toFixed(6)}
+                </span>
+              </div>
+              <div>
+                {runeAmount.toFixed(6)} ($
+                {(runeAmount * runePriceUSD).toFixed(2)})
+              </div>
+            </div>
+
+            <div className="relative mb-4">
+              <Slider
+                value={runeAmount}
+                max={runeBalance}
+                onChange={setRuneAmount}
+              />
+            </div>
+
+            <div className="flex justify-end gap-2">
+              {[25, 50, 100].map((percent) => (
+                <button
+                  key={percent}
+                  className={percentageButtonClasses(
+                    isCloseToPercentage(currentRunePercentage, percent),
+                  )}
+                  onClick={() => handlePercentageClick(percent, true)}
+                >
+                  {percent === 100 ? "MAX" : `${percent}%`}
+                </button>
+              ))}
+            </div>
+          </div>
+        )}
+
+        <Button
+          className="w-full bg-primary text-black font-semibold py-3 rounded-full mt-8"
+          onClick={handleAddLiquidity}
+          disabled={loading || assetAmount <= 0}
+        >
+          {loading ? "Adding Liquidity..." : "Add Liquidity"}
```

Comment:
non-blocking: just going to reiterate here we should display "connect wallet" if the user hasn't connected their wallet yet OR is connecting with a wallet that does not support the asset their depositing. we can do that as a fast follow